core/src/main/java/org/apache/carbondata/core/metadata/SegmentFileStore.java

```diff
@@ -854,7 +853,7 @@ public SegmentFile getSegmentFile() {
    * Gets all index files from this segment
    * @return
    */
-  public Map<String, String> getIndexOrMergeFiles() throws IOException {
+  public Map<String, String> getIndexAndMergeFiles() throws IOException {
```

Comment:
I still think getIndexOrMergeFiles name makes sense. Returned map can have either mergeindex or index. Curious to know when it does have both merge index and index in map ?

Reply:
backward compatibility, carbon 1.3 both index and merge file exist.

Reply:
Or means either, not both, but sometime it will return both index file and merge file, so Or is not correct.

Reply:
Current implementation:
if (null != mergeFileName) {
  add merge file
}
if (null != index file && not empty) {
  add index file
}